### DISCUSSION TITLE
Loading Template in cpp

### DISCUSSION THREAD

**Discussion by andreasschultes** [mikke89/RmlUi#245] [Q&A]
I'm accessing in the following code the private class Rml::StreamFile

```
void Modal::loadTemplate(const std::filesystem::path& templatePath)
{
    auto targetElement = document->GetElementById("content");
    auto* stream = new Rml::StreamFile();
    if (!stream->Open(templatePath))
        return;//TODO error case
    targetElement->SetInnerRML("");
    Rml::Factory::InstanceElementStream(targetElement,stream);
}
```

Is there a better way to load the template? I also thinking using the Template cache is a better way, but is also private.

**Comment by mikke89** [comment]
Hm, could you expand on what you are trying to accomplish here, would it be possible to just load the file and `SetInnerRml(file_contents)`? Since you are going this low level, maybe the template machinery is not really suitable for your use case.
